### ISSUE TITLE
JsonConvert.SerializeObject() incorrectly rounds off double values

### ISSUE THREAD

**Issue by Techiearora7** [JamesNK/Newtonsoft.Json#2540]
The following sample is generated in .NET Core 3.1 environment.

I am trying to serialize a List that contains elements of type double.

To keep things simple I am specifying just a couple of values. 
Source 

0.99470101959419988
6.2202183684300157E-11

### Expected behavior

"Threshold":

0.99470101959419988
6.2202183684300157E-11

### Actual behavior

"Threshold":

0.9947010195941999
6.220218368430016E-11

I have already tried this solution https://github.com/JamesNK/Newtonsoft.Json/issues/2537 but it doesn't seem to work for my case.

**Comment by elgonzo** [comment]
I cannot confirm. Json.Net has no difficulties deserializing 0.99470101959419988 into a double without losing precision.

Proof: https://dotnetfiddle.net/vsOoN6

I guess either your own code has a problem, or is suspect you unbeknownst fell victim to `double.ToString()` doing the rounding thing it does. (To output a `double` with full precision, explicitely call `doubleValue.ToString("G17")`, see also here: https://stackoverflow.com/questions/2105096/why-is-tostring-rounding-my-double-value)

**Comment by elgonzo** [comment]
Yeah, it's unfortunately because Json.NET (incorrectly, in my opinion) uses the roundtrip format specifier "R" for t `double` conversions to strings in the respective JsonConvert methods, like: https://github.com/JamesNK/Newtonsoft.Json/blob/e44ab3348fabcbe9ed6b5053b706a9e10b945ba1/Src/Newtonsoft.Json/JsonConvert.cs#L293-L301
As per documentation at https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings#round-trip-format-specifier-r, "R" can fail to roundtrip certain `double` values. As per the same documentation, it would be better to use the "G17" for `double` and "G9" for `float` instead of "R".

It should be a rather straightforward and quick fix to change the respective float/double-to-string conversion in JsonConvert and elsewhere in the library. However, since this would alter the behavior of the library in some way, i leave it to the author of Json.NET to decide whether and how to address this issue.

As of now, the probably easiest workaround for serializing `double` with full precision is to use a custom JsonConverter that converts `double` values to a string with `.ToString("G17")` and write the converted string value raw to the JsonWriter provided to the converter, kinda like:
```c#
public class FullPrecisionDoubleConverter : JsonConverter<double>
{
    public override bool CanRead => false;

    public override void WriteJson(JsonWriter writer, double value, JsonSerializer serializer)
    {
        var s = value.ToString("G17");
        writer.WriteRawValue(s);
    }

    public override double ReadJson(JsonReader reader, Type objectType, double existingValue, bool hasExistingValue, JsonSerializer serializer)
    {
            throw new NotSupportedException();
    }
}
```
(Dotnetfiddle demoing the converter: https://dotnetfiddle.net/v63Y2f)
